### PR TITLE
Support the "VST SDK Folder" setting

### LINKED ISSUE #157: The "VST SDK Folder" exporter setting is not supported
That setting is only relevant when using JUCE before commit https://github.com/WeAreROLI/JUCE/commit/9f31d6498a899dc3c4f8229bfd203737bc9901a1.

### PR DESCRIPTION
**Pull request by McMartin**
Resolves #157.

@MartyLake: please review, thanks!

### CODE REVIEW

**Comment by McMartin** [comment]
@MartyLake: I totally agree with you that the previous approach had scalability issue. So I reworked the branch to use a list of supported commits (in `commits.hpp`) instead of a switch statement. Please have another look, thanks!

File: Jucer2Reprojucer/main.cpp [hunk 1]
```diff
@@ -145,6 +146,45 @@ int main(int argc, char* argv[])
...
+  const std::string juceCommitSha1 = args.size() == 4 ? args.at(3) : std::string{};
+  const auto commitSha1 = [&juceCommitSha1]() -> unsigned long
+  {
+    if (juceCommitSha1.empty())
+      return kDefaultCommitSha1;
+
+    if (juceCommitSha1.length() < 7)
+      return 0u;
+
+    static_assert(sizeof(unsigned long) >= 7 * 4 / 8, "");
```
Comment (MartyLake) [inline Jucer2Reprojucer/main.cpp]: Missing error message (this sizeof is cryptic, to what command is it associated with ?)

Comment (MartyLake) [inline Jucer2Reprojucer/main.cpp]: the return type of std::stoul ?

Comment (McMartin) [inline Jucer2Reprojucer/main.cpp]: Addressed in f64a2476ec3e1e3f00bd38c6eded1a7039a9f7d0.

**Comment by MartyLake** [comment]
LGTM ! :)


---
Matthieu TALBOT
neogaldr@gmail.com

+33 629 22 15 70
fr.linkedin.com/in/matthieutalbot
http://www.soundcloud.com/Marty-Lake
http://lifehacker.com/5028808/how-eom-makes-your-email-more-efficient

File: Jucer2Reprojucer/main.cpp [hunk 2]
```diff
@@ -145,6 +146,45 @@ int main(int argc, char* argv[])
...
+  {
+    if (juceCommitSha1.empty())
+      return kDefaultCommitSha1;
+
+    if (juceCommitSha1.length() < 7)
+      return 0u;
+
+    try
+    {
+      static_assert(sizeof(decltype(std::stoul(juceCommitSha1))) >= 7 * 4 / 8,
```
Comment (MartyLake) [inline Jucer2Reprojucer/main.cpp]: even better

**Pull request merged**